plume/src/main/kotlin/io/github/plume/oss/passes/type/GlobalTypePass.kt

```diff
@@ -58,4 +70,19 @@ class GlobalTypePass(private val driver: IDriver) : ITypePass {
                 .astParentFullName("<global>").apply { nodeCache.add(this) }
     }
 
-}
\ No newline at end of file
+    private fun getGlobalType(tdFullName: String): NewNodeBuilder {
+        val shortName = if (tdFullName.contains('.')) tdFullName.substringAfterLast('.')
+        else tdFullName
+        return nodeCache.filterIsInstance<NewTypeBuilder>()
+            .find { it.build().properties().get(FULL_NAME).get() == tdFullName }
```

Comment:
Add this to fix the bug, I see the duplication isn't caught since `Type` doesn't have an `AST_PARENT_FULL_NAME` property
```kotlin
            .find { it.build().properties().get(FULL_NAME).get() == tdFullName }
            ?: driver.getVerticesByProperty(FULL_NAME, tdFullName, TYPE).firstOrNull() // <--- this should fix
            ?: NewTypeBuilder()
```